eZ/Publish/Core/FieldType/Keyword/KeywordStorage/Gateway/LegacyStorage.php

```diff
@@ -378,4 +387,54 @@ protected function deleteOrphanedKeywords()
 
         $deleteQuery->prepare()->execute();
     }
+
+    /**
+     * @param \eZ\Publish\Core\Persistence\Database\DatabaseHandler $dbHandler
+     * @param mixed $fieldId
+     * @param int $versionNo
+     * @return bool
+     */
+    private function publishedVersionForFieldExists(
+        DatabaseHandler $dbHandler,
+        $fieldId,
+        $versionNo
+    ) {
+        $selectQuery = $dbHandler->createSelectQuery();
+        $selectQuery->select($dbHandler->quoteColumn('id', 'ezcontentobject_attribute'))
+            ->from(
+                $dbHandler->quoteTable('ezcontentobject_attribute')
+            )
+            ->innerJoin(
+                $dbHandler->quoteTable('ezcontentobject_version'),
+                $dbHandler->quoteColumn('contentobject_id', 'ezcontentobject_attribute'),
+                $dbHandler->quoteColumn('contentobject_id', 'ezcontentobject_version')
+            )
+            ->where(
+                $selectQuery->expr->lAnd(
+                    $selectQuery->expr->eq(
+                        $dbHandler->quoteColumn('id', 'ezcontentobject_attribute'),
+                        $selectQuery->bindValue($fieldId, null, \PDO::PARAM_INT)
+                    ),
+                    $selectQuery->expr->neq(
+                        $dbHandler->quoteColumn('version', 'ezcontentobject_attribute'),
+                        $selectQuery->bindValue($versionNo, null, \PDO::PARAM_INT)
```

Comment:
Edit: I see the `neq` now, that was easy to overlook.
However afaik with content id this query could have been much much simpler right?